contracts/l1-l2-messaging/solidity/src/L1L2.sol

```diff
@@ -26,36 +26,33 @@ contract L1L2Example {
         starknetCore = starknetCore_;
     }
 
-    function get_balance(uint256 user)
+    function get_balance(address user)
         external
         view
         returns (uint256)
     {
         return userBalances[user];
     }
 
-    function withdraw(
-        uint256 l2ContractAddress,
```

Comment:
Consider updating the user's balance before consuming the message from L2 to prevent reentrancy attacks. Additionally, the data type of L2address has been changed to address.